### PR TITLE
Commands: update cmdk

### LINKED ISSUE #59703: Update `cmdk` to latest version and remove temporary fix from PR 56718
### Description

Latest `cmdk` release incorporated a fix for the 'suggestions list' and 'loading progressbar' labels we were waiting for. 
Changelog: https://github.com/pacocoursey/cmdk/releases/tag/v1.0.0
Fix: https://github.com/pacocoursey/cmdk/pull/204

While waiting for this upstream fix we introduced a temporary workaround in https://github.com/WordPress/gutenberg/pull/56718 that can be removed after updating the `cmdk` package and by passing the new two label props. Note we have to pass both new labels because we want them to be translatable.

Cc @youknowriad 

Temporary workaround to be removed:

https://github.com/WordPress/gutenberg/blob/985ea06a811c5ab5d1db31e28d8aba8f57622b02/packages/commands/src/components/command-menu.js#L209-L217

### Step-by-step reproduction instructions

N/A

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #59703

## What?
This PR updates the command package cmdk from version `0.2.0` to `1.0.0`.

At the same time, it removes a temporary workaround for an issue that has been resolved upstream.

## Why?

- An issue with labels was reported in the cmdk library:  https://github.com/pacocoursey/cmdk/issues/196
- A temporary workaround was implemented to avoid this issue: https://github.com/WordPress/gutenberg/pull/56718
- The cmdk library has since fixed the issue and released it as version 1.0.0: https://github.com/pacocoursey/cmdk/pull/204

By updating cmdk to version `1.0.0`, this temporary workaround is no longer necessary.

## How?

Regarding updating the npm packages, I followed the instructions in the article, but please let me know if I'm wrong.

- In the `packages/commands/package.json` file, I manually updated the version of the `cmdk` line to `^1.0.0`.
- Run `npm install`

For labels, I removed the temporary workaround code. Instead, I added a new `label` prop that the List component now supports.

## Testing Instructions

- Open the command palette and enter some text.
- Open your browser developer tools and look at the HTML.
- Please make sure the HTML is properly constructed. The element with the `role="listbox"` attribute should still have the label `Command suggestions` (I've simplified the code for easier reading).

```html
<div class="commands-command-menu__container">
  <div tabindex="-1" cmdk-root="">
    <label cmdk-label="" for=":rg1:" id=":rg0:">Search commands and settings</label>
    <div class="commands-command-menu__header">
      <input placeholder="Search commands and settings" icon="" cmdk-input="" aria-controls=":rfv:" aria-labelledby=":rg0:" id=":rg1:" type="text">
    </div>
    <div cmdk-list="" role="listbox" aria-label="Command suggestions" id=":rfv:"> <!-- 👀 check this element -->
      <div cmdk-list-sizer="">
        <div cmdk-group="" role="presentation" data-value="undefined">
          <div cmdk-group-items="" role="group">
            <div id=":rg7:" cmdk-item="" role="option" aria-disabled="false" aria-selected="true" data-disabled="false" data-selected="true" data-value="Navigation">
              <div class="components-flex components-h-stack commands-command-menu__item has-icon">
                <span>Navigation</span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
```

### CODE REVIEW

File: packages/commands/src/components/command-menu.js [hunk 1]
```diff
@@ -206,16 +206,6 @@ export function CommandMenu() {
...
 
-	// Temporary fix for the suggestions Listbox labeling.
-	// See https://github.com/pacocoursey/cmdk/issues/196
-	useEffect( () => {
-		commandListRef.current?.removeAttribute( 'aria-labelledby' );
-		commandListRef.current?.setAttribute(
-			'aria-label',
-			__( 'Command suggestions' )
-		);
-	}, [ commandListRef.current ] );
```
Comment (youknowriad) [inline packages/commands/src/components/command-menu.js:217]: Nice :)

**Review by youknowriad — APPROVED** [review]
LGTM 👍

File: packages/commands/src/components/command-menu.js [hunk 2]
```diff
@@ -287,7 +277,10 @@ export function CommandMenu() {
 						/>
 						<Icon icon={ inputIcon } />
 					</div>
-					<Command.List ref={ commandListRef }>
+					<Command.List
+						ref={ commandListRef }
```
Comment (afercia) [inline packages/commands/src/components/command-menu.js]: I think this ref can be removed now?
See also line 195.

**Comment by afercia** [comment]
Nice clean up, thanks for taking care of this @t-hamano .
I left one small comment, seems to me a no longer used `ref` can be removed now.

File: packages/commands/src/components/command-menu.js (hunk 2 above)
Comment (t-hamano) [reply packages/commands/src/components/command-menu.js]: Nice catch! Indeed, the ref is no longer necessary. Deleted by [0a0ecc9](https://github.com/WordPress/gutenberg/pull/63465/commits/0a0ecc97ce0b14c1f1a7e536a36659cfcf5e99eb)

**Pull request merged**